src/hdmf/common/resources.py

```diff
@@ -277,7 +306,7 @@ def get_key(self, **kwargs):
             else:
                 return self.keys.row[key_idx_matches[0]]
 
-    @docval({'name': 'resource_name', 'type': str, 'default': None})
+    @docval({'name': 'resource_name', 'type': str, 'doc': 'The name of the resource.'})
```

Comment:
I removed `'default': None`. Why would someone call get_resource without a name? That would just raise an error.